Userland/Shell/AST.cpp

```diff
@@ -3748,11 +3750,13 @@ ErrorOr<String> StringValue::resolve_as_string(RefPtr<Shell> shell)
 ErrorOr<Vector<String>> StringValue::resolve_as_list(RefPtr<Shell> shell)
 {
     if (is_list()) {
+        // FIXME: Catch the allocation failure from StringView::split_view, by using
+        //        String::split(String const&,...)
```

Comment:
That function is not a static function, why are you naming it like one?
also the fixme sounds superfluous, what's stopping this from happening?

Reply:
The API is not there yet, and I could not be bothered to do a full rebuild, with changed String headers

Reply:
Ideally we would use `for_each_split_view`, but then its either multiple allocations from append, or virtually two splits (which actually shouldn't be that expensive)
You can make a call and I'll change it accordingly

Reply:
:thinking: 
sure is, `String.h:148`
```c++
    ErrorOr<Vector<String>> split(u32 separator, SplitBehavior = SplitBehavior::Nothing) const;
```

Reply:
oh we're splitting by a string, got it

Reply:
I now changed it to MaxDue's